x-pack/plugins/lens/public/editor_frame_service/format_column.ts

```diff
@@ -69,25 +67,27 @@ export const formatColumn: ExpressionFunctionDefinition<
       types: ['number'],
       help: '',
     },
-    nestedFormat: {
+    rangeParameters: {
       types: ['string'],
       help: '',
     },
-    template: {
-      types: ['string'],
-      help: '',
-    },
-    replaceInfinity: {
-      types: ['boolean'],
-      help: '',
-    },
   },
-  inputTypes: ['kibana_datatable'],
-  fn(input, { format, columnId, decimals, nestedFormat, template, replaceInfinity }: FormatColumn) {
+  inputTypes: ['datatable'],
+  fn(input, { format, columnId, decimals, rangeParameters }: FormatColumn) {
     return {
       ...input,
       columns: input.columns.map((col) => {
         if (col.id === columnId) {
+          const {
+            template,
+            replaceInfinity,
+            nestedFormat,
+          }: {
+            template?: string;
+            replaceInfinity?: boolean;
+            nestedFormat?: string;
+          } = rangeParameters ? JSON.parse(rangeParameters) : {};
```

Comment:
Is there a way to abstract this one more level? Ideally the editor_frame_service would contain code that has no references to any specific datasource.

If it's not possible to abstract this another level then maybe we should move the format_column code into the datasource.

Reply:
I used the same approach as in the https://github.com/elastic/kibana/blob/master/x-pack/plugins/lens/public/indexpattern_datasource/rename_columns.ts#L39 .
From my understanding this function is called from the expression service. Can you confirm?

I can agree on the `format_column` move at datasource level, but probably I'd do in a separate PR.